### PR TITLE
Rewrite states and some fixes regarding the latest commits

### PR DESCRIPTION
**Pull request by j1s2e3**

### CODE REVIEW

File: lib/definitions/eep/TF-13/TF-13-21.json [hunk 1]
```diff
@@ -291,48 +301,28 @@
                     "shortcut": "DR",
                     "bitoffs": "15",
                     "bitsize": "1",
-                    "value_out": {
-                        "if": [
-                            {"==":[false, {"var": "value"}]}, 0,
-                            {"==":[true, {"var": "value"}]}, 1
-                        ]
-                    }
+                    "value_out": {"if":[{"var":"value"},1,0]}
```
Comment (Jey-Cee) [inline lib/definitions/eep/TF-13/TF-13-21.json:304]: This will not work
Please check out https://jsonlogic.com/operations.html#if

File: lib/definitions/eep/TF-13/TF-13-20.json [hunk 2]
```diff
@@ -82,14 +102,7 @@
                 },
                 {
                     "data": "fixed parameter",
-                    "description": "dummy",
```
Comment (Jey-Cee) [inline lib/definitions/eep/TF-13/TF-13-20.json:85]: Each byte of a telegramm has to be set, otherwise the payload length could be false

File: lib/definitions/eep/TF-13/TF-13-07.json [hunk 3]
```diff
@@ -53,39 +65,16 @@
...
-                },
-                {
-                    "data": "Dimming speed",
-                    "shortcut": "DS",
-                    "bitoffs": "16",
-                    "bitsize": "8",
-                    "value": 0
-                }
-            ]
-        },
```
Comment (Jey-Cee) [inline lib/definitions/eep/TF-13/TF-13-07.json:88]: Why did you delete this part?

File: lib/definitions/eep/TF-13/TF-13-21.json (hunk 1 above)
Comment (j1s2e3) [reply lib/definitions/eep/TF-13/TF-13-21.json:304]: I have this type of if statement out of [https://jsonlogic.com/operations.html#if](https://jsonlogic.com/operations.html#if).
I came along with this solution, trying to implement the Setpoint Priority for the FTAF55D
It's tested and it works for me without any problems.

It's the first in the above doc: {"if" : [ true, "yes", "no" ]}
works like the ? operator in C.
char *string;
...
printf("string=%s\n", string ? string : "NULL");
 
Step by step looking at "DR - Dim red":
        "DR": {
            "type": "state",
            "common": {
                "name": "Dim red",
                "role": "state",
                **"type": "boolean",**
                "read": true,
                "write": true,
                "def": false
            },
            "native": {}
        },
DR is already a boolean.

From the send (value=4) case:
                {
                    "data": "Dim red",
                    "shortcut": "DR",
                    "bitoffs": "15",
                    "bitsize": "1",
                    "value_out": {"if":[**{"var":"value"}**,1,0]}
                },
{"var":"value"} evaluates to "true" or "false" depending the value of the datapoint DR.
DR=true {"if" : [ true, 1, 0 ]} => value_out=1
DR=false {"if" : [ false, 1, 0 ]} => value_out=0
In pseudo code;
if DR then
    value_out = 1
else
    value_out = 0
endif

That easy and quicker as the larger variant:
"value_out": {
                        "if": [
                            {"==":[false, {"var": "value"}]}, 0,
                            {"==":[true, {"var": "value"}]}, 1
                        ]
                    }

...which evaluates to something like this:
if DR==false then
    value_out=0
else
    if DR==true then
        value_out=1
    endif
endif

And as I said, I have tested it. I always test stuff before I commit anything. It works for the FTAF55D and also for the FRGBW71L master. Play around with DR/DG/DB/DW, send a DimUp and look at the telegram. The bits are going to be set correct. Same for the setpoint pri bit in the FTAF telegramm.

File: lib/definitions/eep/TF-13/TF-13-07.json (hunk 3 above)
Comment (j1s2e3) [reply lib/definitions/eep/TF-13/TF-13-07.json:88]: It's not deleted. Just moved to the end, so that all 6 sends are together and the read part comes last. It's just cosmetic, easier to read.

File: lib/definitions/eep/TF-13/TF-13-20.json (hunk 2 above)
Comment (j1s2e3) [reply lib/definitions/eep/TF-13/TF-13-20.json:85]: OK, understand. In the README there is nothing documented how to setup a telegram going to be sent. So I was learning by doing, reading examples and trying out different approaches. It seems, that the telegram is initialized with all zeros, and then you can poke in the bits and bytes of interest. At least it works. So I thought poking in zeros is redundant.

The existing EEPs are not straightforward either. So it's not really clear for me, how to setup a telegram.

For instance. EEP TF-13-21. The DR/DG/DB/DW bits. Bitoffset 12-15 are set from the datapoints and bits 8-11 are set to zero. So according to this example, **every** bit of the telegram is being touched.

On the other hand, in EEP TF-13-07, send (value 0) not every bit is touched, not even every byte???

So I made some test of the behavior and figured out, that zeros don't have to be initialized.

Solution: revise all EEPs and ensure that all bits are touched or assure that the payload length is always correct, even if only bits/bytes of interest of touched.

File: lib/definitions/eep/TF-13/TF-13-21.json (hunk 1 above)
Comment (Jey-Cee) [reply lib/definitions/eep/TF-13/TF-13-21.json:304]: Testing means nothing. I tried it on the site and result was always 0.
There was a syntax error at the data site, which was not marked so.

Comment (j1s2e3) [reply lib/definitions/eep/TF-13/TF-13-21.json:304]: EEP TF-13-20, FTAF55D, source from eep-fixes-states branch:

```
...
        "SPRI": {
            "type": "state",
            "common": { 
                "name": "Set point priority",
                "role": "state",
                "type": "boolean",
                "read": true,
                "write": true,
                "def": true
            },
            "native": {}
        }
...
        {
            "send": true,
            "condition": { 
                "command": [
                    {   
                        "value": 0
                    }
                ]
            },
            "datafield": [
                {
                    "data": "Set point 12...28°C",
                    "shortcut": "SP",
                    "bitoffs": "8",
                    "bitsize": "8",
                    "value_out": {"+":[{"/":[{"-":[{"var":"value"},0]},0.1568627450980392]},0]}
                },
                {
                    "data": "fixed parameter",
                    "description": "Must be set",
                    "bitoffs": "28",
                    "bitsize": "1",
                    "value": 1
                },
                {
                    "data": "Set point priority",
                    "shortcut": "SPRI",
                    "bitoffs": "30",
                    "bitsize": "1",
                    "value_out": {"if":[{"var":"value"},0,1]}
                }
            ]
        },
```

Under objects: SP=20, SPRI=**true**, CMD=Send data:
Sent data: 55000a0701eb a5007f000**8** .src-id. 0000 .dst-id. ff00e5

Under objects: SP=20, SPRI=**false**, CMD=Send data:
Sent data: 55000a0701eb a5007f000**a** .src-id. 0000 .dst-id. ff00ca

Under objects: SP=20, SPRI=**true**, CMD=Send data:
Sent data: 55000a0701eb a5007f000**8** .src-id. 0000 .dst-id. ff00e5

Comment (j1s2e3) [reply lib/definitions/eep/TF-13/TF-13-21.json:304]: EEP TF13-21, FRGBW71L master, source from eep-fixes-states branch:
```
        {
            "send": true,
            "condition": {
                "command": [
                    {   
                        "value": 4
                    }
                ]
            },
            "datafield": [
                {   
                    "data": "Dim speed",
                    "shortcut": "DS",
                    "bitoffs": "0",
                    "bitsize": "8",
                    "value_out": {"var": "value"}
                },
                {   
                    "data": "Dim red",
                    "shortcut": "DR",
                    "bitoffs": "15",
                    "bitsize": "1",
                    "value_out": {"if":[{"var":"value"},1,0]}
                },
                {
                    "data": "Dim green",
                    "shortcut": "DG",
                    "bitoffs": "14",
                    "bitsize": "1",
                    "value_out": {"if":[{"var":"value"},1,0]}
                },
                {
                    "data": "Dim blue",
                    "shortcut": "DB",
                    "bitoffs": "13",
                    "bitsize": "1",
                    "value_out": {"if":[{"var":"value"},1,0]}
                },
                {   
                    "data": "Dim white",
                    "shortcut": "DW",
                    "bitoffs": "12",
                    "bitsize": "1",
                    "value_out": {"if":[{"var":"value"},1,0]}
                },
                {
                    "data": "fixed parameter",
                    "description": "Not used",
                    "bitoffs": "8",
                    "bitsize": "4",
                    "value": 0
                },
                {
                    "data": "fixed parameter",
                    "description": "Dim up",
                    "bitoffs": "16",
                    "bitsize": "8",
                    "value": 48
                },
                {
                    "data": "fixed parameter",
                    "description": "GFVS",
                    "bitoffs": "24",
                    "bitsize": "8",
                    "value": 15
                }
            ]
        },
```

Under objects: DS=128, DR=false, DG=false, DB=false, DW=false, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**0**300f .src-id. 0000 .dst-id. ff00a1

Under objects: DS=128, DR=true, DG=false, DB=false, DW=false, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**1**300f .src-id. 0000 .dst-id. ff0014

Under objects: DS=128, DR=false, DG=true, DB=false, DW=false, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**2**300f .src-id. 0000 .dst-id. ff00cc

Under objects: DS=128, DR=false, DG=false, DB=true, DW=false, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**4**300f .src-id. 0000 .dst-id. ff007b

Under objects: DS=128, DR=false, DG=false, DB=false, DW=true, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**8**300f .src-id. 0000 .dst-id. ff0012

Under objects: DS=128, DR=true, DG=true, DB=true, DW=true, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**f**300f .src-id. 0000 .dst-id. ff0010

Under objects: DS=128, DR=false, DG=true, DB=true, DW=true, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**e**300f .src-id. 0000 .dst-id. ff00a5

Under objects: DS=128, DR=true, DG=false, DB=true, DW=true, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**d**300f .src-id. 0000 .dst-id. ff007d

Under objects: DS=128, DR=true, DG=true, DB=false, DW=true, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**b**300f .src-id. 0000 .dst-id. ff00ca

Under objects: DS=128, DR=true, DG=true, DB=true, DW=false, CMD=Dim up(4)
Sent data: 55000a0701eb a5800**7**300f .src-id. 0000 .dst-id. ff00a3

Comment (j1s2e3) [reply lib/definitions/eep/TF-13/TF-13-21.json:304]: I did not meant testing on the json logic site, but in ioBroker itself. I use this type of logic in the TF-13-20/21 EEPs. Above are the test results for both EEPs. As you can see, it works fine. It's a standart, up-to-date ioBroker installion on a raspberry pi 4 (“curl -sL https://iobroker.net/install.sh | bash -”). A fresh install done 6 weeks ago.

I don't like to have endless discussions here to convince you that "if":[{"var":"value"},1,0] do work. I you don't like it go ahead and stay with "if":[{"==":[false,{"var":"value"}]},0,{"==":[true,{"var":"value"}]},1]. It's your repositoy.

All I want is a working repository for my home setup.
